substrate/executor/src/native_executor.rs

```diff
@@ -88,17 +115,25 @@ impl<D: NativeExecutionDispatch + Sync + Send> CodeExecutor for NativeExecutor<D
 		method: &str,
 		data: &[u8],
 	) -> Result<Vec<u8>> {
-		if code == D::native_equivalent() {
-			// call native
-			D::dispatch(ext, method, data)
-		} else {
-			let version = WasmExecutor.call(ext, code, "version", &[])?;
-			let version = RuntimeVersion::decode(&mut version.as_slice());
-			if version.map_or(false, |v| D::VERSION.can_call_with(&v)) {
-				return D::dispatch(ext, method, data)
-			}
-			// call into wasm.
-			WasmExecutor.call(ext, code, method, data)
+		match RUNTIMES_CACHE.lock().entry(gen_cache_key(code)).or_insert_with(|| {
+					let wasm_module = WasmModule::from_buffer(code)
+																.expect("all modules compiled with rustc are valid wasm code; qed");
```

Comment:
indent of run-on lines should be one more than original

Reply:
do we have a written style guide somewhere?

Reply:
https://github.com/paritytech/polkadot/wiki/Style-Guide